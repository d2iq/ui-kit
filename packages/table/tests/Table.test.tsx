import React from "react";

import { Table, Column } from "../";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import { render, shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("Table", () => {
  const items = [
    {
      name: "Brian Vaughn",
      role: "Software Engineer",
      city: "San Jose",
      state: "CA",
      zipcode: 95125
    },
    {
      name: "Jon Doe",
      role: "Product engineer",
      city: "Mountain View",
      state: "CA",
      zipcode: 95125
    }
  ];

  const nameCellRenderer = ({ name }: { name?: string }) => (
    <strong>{name}</strong>
  );
  const cityCellRenderer = ({ city }: { city?: string }) => (
    <strong>{city}</strong>
  );
  const roleCellRenderer = ({ role }: { role?: string }) => <em>{role}</em>;
  const stateCellRenderer = ({ state }: { state?: string }) => (
    <span>{state}</span>
  );
  const zipcodeCellRenderer = ({ zipcode }: { zipcode?: string }) => (
    <span>{zipcode}</span>
  );
  const width = ({ width: totalWidth }: { width?: number }) =>
    totalWidth ? totalWidth * 0.3 : 100;

  it("renders default", () => {
    const component = render(
      <Table data={items}>
        <Column header="name" cellRenderer={nameCellRenderer} width={width} />
        <Column header="role" cellRenderer={roleCellRenderer} width={width} />
        <Column header="state" cellRenderer={stateCellRenderer} width={width} />
        <Column
          header="zipcode"
          cellRenderer={zipcodeCellRenderer}
          width={width}
        />
        <Column header="city" cellRenderer={cityCellRenderer} width={width} />
      </Table>
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("renders again with new data", () => {
    const itemRenderer = item => <span>{item}</span>;
    const component = mount(
      <Table data={[1, 2, 3]}>
        <Column header="item" cellRenderer={itemRenderer} width={width} />
      </Table>
    );
    expect(toJson(component)).toMatchSnapshot();

    component.setProps({
      data: [4, 5, 6]
    });
    expect(toJson(component)).toMatchSnapshot();
  });

  describe("getData", () => {
    it("will return the right array", () => {
      const width = () => 10;
      const cell = number => <span>{number}</span>;
      const component = shallow(
        <Table data={[1, 2, 3]}>
          <Column header="test" cellRenderer={cell} width={width} />
        </Table>
      );
      const instance = component.instance() as Table<number>;
      expect(instance.getData([1, 2, 3])).toEqual([{}, 1, 2, 3, {}]);
    });
  });
});
