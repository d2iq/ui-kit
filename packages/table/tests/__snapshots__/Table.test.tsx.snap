// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders again with new data 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  font-weight: normal;
}

<Table
  data={
    Array [
      1,
      2,
      3,
    ]
  }
>
  <AutoSizer
    __dataThatTriggersARerenderWhenChanged={
      Array [
        1,
        2,
        3,
      ]
    }
    className="emotion-0"
    defaultHeight={768}
    defaultWidth={1024}
    disableHeight={false}
    disableWidth={false}
    onResize={[Function]}
    style={Object {}}
  >
    <div
      className="emotion-0"
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <MultiGrid
        cellRenderer={[Function]}
        classNameBottomLeftGrid=""
        classNameBottomRightGrid=""
        classNameTopLeftGrid=""
        classNameTopRightGrid=""
        columnCount={1}
        columnWidth={[Function]}
        enableFixedColumnScroll={true}
        enableFixedRowScroll={true}
        fixedColumnCount={1}
        fixedRowCount={1}
        height={0}
        hideBottomLeftGridScrollbar={true}
        hideTopRightGridScrollbar={true}
        rowCount={5}
        rowHeight={35}
        scrollToColumn={-1}
        scrollToRow={-1}
        style={Object {}}
        styleBottomLeftGrid={Object {}}
        styleBottomRightGrid={Object {}}
        styleTopLeftGrid={Object {}}
        styleTopRightGrid={Object {}}
        width={0}
      />
    </div>
  </AutoSizer>
</Table>
`;

exports[`Table renders again with new data 2`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  font-weight: normal;
}

<Table
  data={
    Array [
      4,
      5,
      6,
    ]
  }
>
  <AutoSizer
    __dataThatTriggersARerenderWhenChanged={
      Array [
        4,
        5,
        6,
      ]
    }
    className="emotion-0"
    defaultHeight={768}
    defaultWidth={1024}
    disableHeight={false}
    disableWidth={false}
    onResize={[Function]}
    style={Object {}}
  >
    <div
      className="emotion-0"
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <MultiGrid
        cellRenderer={[Function]}
        classNameBottomLeftGrid=""
        classNameBottomRightGrid=""
        classNameTopLeftGrid=""
        classNameTopRightGrid=""
        columnCount={1}
        columnWidth={[Function]}
        enableFixedColumnScroll={true}
        enableFixedRowScroll={true}
        fixedColumnCount={1}
        fixedRowCount={1}
        height={0}
        hideBottomLeftGridScrollbar={true}
        hideTopRightGridScrollbar={true}
        rowCount={5}
        rowHeight={35}
        scrollToColumn={-1}
        scrollToRow={-1}
        style={Object {}}
        styleBottomLeftGrid={Object {}}
        styleBottomRightGrid={Object {}}
        styleTopLeftGrid={Object {}}
        styleTopRightGrid={Object {}}
        width={0}
      />
    </div>
  </AutoSizer>
</Table>
`;

exports[`Table renders default 1`] = `
.emotion-12 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  font-weight: normal;
}

.emotion-0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-top: 1px solid #000;
  border-bottom: 1px solid #000;
  height: 35px;
  left: 0;
  position: absolute;
  top: 0;
  width: 307.2px;
}

.emotion-2 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-top: 1px solid #000;
  border-bottom: 1px solid #000;
  height: 35px;
  left: 307.2px;
  position: absolute;
  top: 0;
  width: 307.2px;
}

.emotion-3 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-top: 1px solid #000;
  border-bottom: 1px solid #000;
  height: 35px;
  left: 614.4px;
  position: absolute;
  top: 0;
  width: 307.2px;
}

.emotion-4 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-bottom: 1px solid #DADDE2;
  height: 35px;
  left: 0;
  position: absolute;
  top: 0;
  width: 307.2px;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-bottom: 1px solid #DADDE2;
  height: 35px;
  left: 0;
  position: absolute;
  top: 35px;
  width: 307.2px;
}

.emotion-7 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-bottom: 1px solid #DADDE2;
  height: 35px;
  left: 307.2px;
  position: absolute;
  top: 0;
  width: 307.2px;
}

.emotion-8 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-bottom: 1px solid #DADDE2;
  height: 35px;
  left: 614.4px;
  position: absolute;
  top: 0;
  width: 307.2px;
}

.emotion-10 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-bottom: 1px solid #DADDE2;
  height: 35px;
  left: 307.2px;
  position: absolute;
  top: 35px;
  width: 307.2px;
}

.emotion-11 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Helvetica','Arial',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol';
  border-bottom: 1px solid #DADDE2;
  height: 35px;
  left: 614.4px;
  position: absolute;
  top: 35px;
  width: 307.2px;
}

<div
  class="emotion-12"
  style="overflow:visible;height:0;width:0"
>
  <div
    style="height:768px;overflow:visible;width:1024px"
  >
    <div
      style="height:35px;position:relative;width:1024px"
    >
      <div
        aria-label="grid"
        aria-readonly="true"
        class="ReactVirtualized__Grid"
        role="grid"
        style="box-sizing:border-box;direction:ltr;height:35px;position:absolute;width:307.2px;-webkit-overflow-scrolling:touch;will-change:transform;overflow-x:hidden;overflow-y:hidden;left:0;top:0"
      >
        <div
          class="ReactVirtualized__Grid__innerScrollContainer"
          role="rowgroup"
          style="width:307.2px;height:35px;max-width:307.2px;max-height:35px;overflow:hidden;pointer-events:;position:relative"
        >
          <div
            class="emotion-0"
          >
            name
          </div>
        </div>
      </div>
      <div
        class="TopRightGrid_ScrollWrapper"
        style="left:307.2px;overflow-x:hidden;overflow-y:hidden;position:absolute;top:0;height:35px;width:716.8px"
      >
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid"
          role="grid"
          style="box-sizing:border-box;direction:ltr;height:35px;position:absolute;width:716.8px;-webkit-overflow-scrolling:touch;will-change:transform;overflow-x:auto;overflow-y:hidden;left:0;top:0"
        >
          <div
            class="ReactVirtualized__Grid__innerScrollContainer"
            role="rowgroup"
            style="width:1021.5999999999999px;height:35px;max-width:1021.5999999999999px;max-height:35px;overflow:hidden;pointer-events:;position:relative"
          >
            <div
              class="emotion-0"
            >
              role
            </div>
            <div
              class="emotion-2"
            >
              state
            </div>
            <div
              class="emotion-3"
            >
              zipcode
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="height:733px;overflow:visible;position:relative;width:1024px"
    >
      <div
        class="BottomLeftGrid_ScrollWrapper"
        style="left:0;overflow-x:hidden;overflow-y:hidden;position:absolute;height:733px;width:307.2px"
      >
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid"
          role="grid"
          style="box-sizing:border-box;direction:ltr;height:733px;position:absolute;width:307.2px;-webkit-overflow-scrolling:touch;will-change:transform;overflow-x:hidden;overflow-y:auto;left:0"
        >
          <div
            class="ReactVirtualized__Grid__innerScrollContainer"
            role="rowgroup"
            style="width:307.2px;height:105px;max-width:307.2px;max-height:105px;overflow:hidden;pointer-events:;position:relative"
          >
            <div
              class="emotion-4"
            >
              <strong>
                Brian Vaughn
              </strong>
            </div>
            <div
              class="emotion-5"
            >
              <strong>
                Jon Doe
              </strong>
            </div>
            <div
              style="height:35px;left:0;position:absolute;top:70px;width:307.2px"
            >
              Â 
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="grid"
        aria-readonly="true"
        class="ReactVirtualized__Grid"
        role="grid"
        style="box-sizing:border-box;direction:ltr;height:733px;position:absolute;width:716.8px;-webkit-overflow-scrolling:touch;will-change:transform;overflow-x:auto;overflow-y:hidden;left:307.2px"
        tabindex="0"
      >
        <div
          class="ReactVirtualized__Grid__innerScrollContainer"
          role="rowgroup"
          style="width:1021.5999999999999px;height:105px;max-width:1021.5999999999999px;max-height:105px;overflow:hidden;pointer-events:;position:relative"
        >
          <div
            class="emotion-4"
          >
            <em>
              Software Engineer
            </em>
          </div>
          <div
            class="emotion-7"
          >
            <span>
              CA
            </span>
          </div>
          <div
            class="emotion-8"
          >
            <span>
              95125
            </span>
          </div>
          <div
            class="emotion-5"
          >
            <em>
              Product engineer
            </em>
          </div>
          <div
            class="emotion-10"
          >
            <span>
              CA
            </span>
          </div>
          <div
            class="emotion-11"
          >
            <span>
              95125
            </span>
          </div>
          <div
            style="height:35px;left:0;position:absolute;top:70px;width:307.2px"
          >
            Â 
          </div>
          <div
            style="height:35px;left:307.2px;position:absolute;top:70px;width:307.2px"
          >
            Â 
          </div>
          <div
            style="height:35px;left:614.4px;position:absolute;top:70px;width:307.2px"
          >
            Â 
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
